### DISCUSSION TITLE
My automated twitter cards are not showing up on Twitter?

### DISCUSSION THREAD

**Discussion by thegilpatrick** [quarto-dev/quarto-cli#5973] [Q&A] [answered]
### Description

Hey,

I just created a blog via Quarto and tried enabling the automated twitter cards for my posts. 

Here is my shortened _quarto.yaml.
````qmd
website:
  favicon: thegilpatricklogo.png
  open-graph: true
  twitter-card: true
  site-url: https://thegilpatrick.com
````

When I try to post a link to a blog post webpage for example: https://thegilpatrick.com/posts/recipe-web-scrape/ Nothing shows up on Twitter. Even though when I view the page source search the html I found the twitter card stuff near the bottom of the page.

```
<meta property="og:title" content="Recipe Web Scraping in Observable">
<meta property="og:description" content="Tutorial on how to web scrape recipes with Observable.">
<meta property="og:image" content="https://thegilpatrick.com/posts\recipe-web-scrape\recipe.webp">
<meta property="og:site-name" content="The Gilpatrick">
<meta property="og:image:alt" content="Zucchini Courgette">
<meta name="twitter:title" content="Recipe Web Scraping in Observable">
<meta name="twitter:description" content="Tutorial on how to web scrape recipes with Observable.">
<meta name="twitter:image" content="https://thegilpatrick.com/posts\recipe-web-scrape\recipe.webp">
<meta name="twitter:image:alt" content="Zucchini Courgette">
<meta name="twitter:card" content="summary_large_image">
```

Why do you think the Twitter web crawler cannot find it? Could it have to do with it being too far down the page starting on line 25304?

**Comment by dragonstyle** [comment]
I ran the URL through their validator and it complained about the size being too large:

https://cards-dev.twitter.com/validator

I'm not sure what is making the page excessively large though - not sure if that is a helpful clue. I also noticed that the image path itself is mixing forward and backward slashes, which may work for twitter but is clearly a bug in Quarto. What version of Quarto are you using - I feel like we may have fixed this but would be good to confirm that you're on the latest released build of Quarto 1.3.

**Reply by thegilpatrick** [reply]
My version of Quarto is 1.3.159

**Reply by dragonstyle** [reply]
Ok that is definitely recent enough that it seems bad to be mixing slashes! I'll have a look at that thank you!

**Reply by dragonstyle** [reply]
I just tried to reproduce this on windows and wasn't able to (using the latest version of Quarto). I am always seeing valid well formed paths for the Twitter cards when I clone your repo and build it locally :(.

**Reply by thegilpatrick** [reply]
Does it have something to do with how I publish it on GitHub Pages? I used the docs method outlined here https://quarto.org/docs/publishing/github-pages.html

I thought maybe enabling the `embed-resources: true` option was the problem with the file size.

**Reply by mcanouil** [reply]
Why do you use `embed-resources: true` for a website? This means the libs are embedded possibly in each pages.
This could explain why Twitter and OpenGraph fail to retrieve the information.

Also, I would suggest to at least use the latest 1.3 release, as there were few issues related to these metadata that have been fixed.

> Does it have something to do with how I publish it on GitHub Pages? I used the docs method outlined here https://quarto.org/docs/publishing/github-pages.html

Hum, your repository does not seem to follow any of the documentation, _i.e._, you have a directory with the source code which you render at the root of the repository.

**Reply by thegilpatrick** [reply]
I am rendering the website via the VSCode Quarto extension if that has anything to do with it? Otherwise I guess I can just edit the hyperlinks in the outputted html files for now? Until I figure out what is going on.

**Reply by thegilpatrick** [reply]
Sorry yes that is right @mcanouil. 

When I tried rendering the output to the docs directory and pushing that to github. I could not get GitHub pages to render my .css files properly. 

I ended up having to do the self-contained: true to get the styling to work. Then I just dragged and dropped the html files from the output as well as my source code for the .qmd files so people could see them if they want.

**Reply by mcanouil** [reply]
I forked your repository to try it and I can't render at all.
There a lot of missing resources in your repository:
- subscribe.html
- no-title.css
- styles.css
- ./thegilpatricklogo.png


> I ended up having to do the self-contained: true to get the styling to work. Then I just dragged and dropped the html files from the output as well as my source code for the .qmd files so people could see them if they want.

This is likely why there are issues in your website.

**Reply by mcanouil** [reply]
I highly recommend, you to follow the documentation on how to start a website and how to deploy the website to GitHub Pages.

With the current state of your repository, it's unlikely we can figure out what's wrong with social media cards amongst everything else.

Once, you get a clean repository that we can render, if after reading the documentation on publishing your website you still have issues to publish it, please do come back to us.

**Reply by thegilpatrick** [reply]
Sounds good. Just to be clear is there a reason the standalone html files can't use OpenGraph or Twitter Cards yet?

**Reply by mcanouil** [reply]
As I said, there are currently many things wrong in your repository not allowing us to render your project, thus we can't investigate properly.

**Reply by thegilpatrick** [reply]
That's fair. I will try to publish tomorrow and turn off the `embed-resources: true` option. I think you are right that is creating the file size problem.

**Reply by mcanouil** [reply]
There is also no use to do that for website, it increases a lot the loading time, since libs are no longer shared by multiple pages but embedded in each.
`embed-resources` is useful when you want to send the file, because you can send one file and not an archive.

1. Cleanup your repository
2. Change the output directory as explained in the documentation (see <https://quarto.org/docs/publishing/github-pages.html#render-to-docs>).
   ```yaml
   project:
     type: website
     output-dir: docs
   ```
3. Render your website
4. Go to your repository page setting and select the proper branch and directory
<img width="801" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/25ee9837-0cd9-4bce-bcc7-6052e7dce4fb">
5. Enjoy your website on GitHub Pages

**Comment by thegilpatrick** [comment]
Okay I republished to Github Pages without `embed-resources: true` and the twitter cards are still not working.

**Reply by thegilpatrick** [reply]
Here is the repository: https://github.com/thegilpatrick/Blog

**Reply by mcanouil** [reply, marked as answer]
You still have issues in your website:
- I add to exclude the R Tidycensus post
- you still have pages with `embed-resources: true`
- your `site-url` is wrong for some reason (probably a bad setup of your domain in Netlify): <https://www.opengraph.xyz/url/https%3A%2F%2Fthegilpatrick.com%2Fposts%2Frecipe-web-scrape%2F>
- Also you still have weird backslashes in the URLs which I can't explain (Windows?).

See <https://www.opengraph.xyz/url/https%3A%2F%2Fm.canouil.dev%2Fquarto-issue5973%2Fposts%2Frecipe-web-scrape%2F> from my fork at <https://github.com/mcanouil/quarto-issue5973> and published to GitHub Pages (<https://m.canouil.dev/quarto-issue5973/>).
For short, it is working "fine" for me.

What's the output of `quarto check`?

EDIT: reproducible example/fork moved to <https://github.com/mcanouil/quarto-issues/tree/main/quarto-cli-5973>

**Reply by thegilpatrick** [reply]
>your site-url is wrong for some reason (probably a bad setup of your domain in Netlify):

Yeah sorry I tried publishing with Netlify earlier and was in the process of switching the domain over to Netlify. But then eventually went back to publishing with Github pages. I need to re-render to the docs directory.

>See https://www.opengraph.xyz/url/https%3A%2F%2Fm.canouil.dev%2Fquarto-issue5973%2Fposts%2Frecipe-web-scrape%2F from my fork at https://github.com/mcanouil/quarto-issue5973 and published to GitHub Pages (https://m.canouil.dev/quarto-issue5973/).
For short, it is working "fine" for me.

Now I can see the cards but the image preview is still not showing in your forked version. Any guesses as to why that would be?

**Reply by mcanouil** [reply]
What do you mean by "image preview"?
I updated the site-url since I renamed the fork.
<img width="1503" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/002e22ed-11b3-4026-94ea-16a1172d8d86">

**Reply by thegilpatrick** [reply]
This is how it shows up when I link it in Twitter
![no-img-twitter-card](https://github.com/quarto-dev/quarto-cli/assets/61705820/7321ac30-8af8-4d6d-b3f1-48718fe4522e)

Here is how a normal twitter card looks when I link in Twitter.
![img-twitter-card](https://github.com/quarto-dev/quarto-cli/assets/61705820/b21d9db6-2616-428c-a710-84be35d6fdb6)

**Reply by mcanouil** [reply]
You need to wait for the DNS cache to propagate.

It's working fine on LinkedIn
<img width="1487" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/2575d326-0778-4163-8059-b9d1bdf7cf9c">

And working fine on Twitter
<img width="1500" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/aad153e3-ed98-4031-b563-1369d13de0bf">

**Reply by thegilpatrick** [reply]
Sweet! Okay thanks! I will make sure to fix the things you noted.
